app/src/main/java/com/globant/harrypotterapp/util/Data.kt

```diff
@@ -0,0 +1,9 @@
+package com.globant.harrypotterapp.util
+
+data class Data<RequestData>(var status: Status, var data: RequestData? = null, var error: Exception? = null)
```

Comment:
which is the use of this nullables? can this have more than one constructor?

Reply:
The use of this nullables is when I need to set different states in the viewmodels LiveData, because sometimes I need to set a "data" value if the API response is Success or a "error" value when de API response is Failure. Also I need to set just a "status" value when the viewmodel starts to fetch the data from de API and the state is "LOADING".  The instances of this class depends on what the API response returns to the viewmodel, so yes this can have more than one constructor.